src/controller/content/content.service.ts

```diff
@@ -1853,4 +1912,180 @@ export class ContentService {
   }
 
 
+  async getAllQuestionTypes() {
+    try {
+      const allQuestionTypes = await db.select().from(zuvyQuestionTypes);
+      if (allQuestionTypes.length > 0) {
+        return {
+          status: 'success',
+          code: 200,
+          allQuestionTypes,
+        };
+      } else {
+        return [];
+      }
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  async createQuestionType(questionType: CreateTypeDto) {
+    try {
+      const newQuestionType = await db.insert(zuvyQuestionTypes).values(questionType).returning();
+      if (newQuestionType.length > 0) {
+        return {
+          status: 'success',
+          code: 200,
+          newQuestionType,
+        };
+      } else {
+        return {
+          status: 'error',
+          code: 404,
+          message: 'Question Type is not created.Please try again.',
+        };
+      }
+    } catch (err) {
+      throw err;
+    }
+  }
+
+
+  async createFormForModule(form: formBatchDto) {
+    try {
+      const formQuestions = form.questions.map((f) => ({
+        question: f.question,
+        options: f.options,
+        typeId: f.typeId,
+      }));
+
+      const result = await db
+        .insert(zuvyModuleForm)
+        .values(formQuestions)
+        .returning();
+      if (result.length > 0) {
+        return {
+          status: "success",
+          code: 200,
+          result
+        }
+      }
+      else {
+        return {
+          status: "error",
+          code: 404,
+          message: "Form questions did not create successfully.Please try again"
+        }
+      }
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  async getAllFormQuestions(
+    typeId: number,
+    searchTerm: string = '',
+  ) {
+    try {
+      let queryString;
+      if (!Number.isNaN(typeId) && questionType == undefined) {
+        queryString = sql`${zuvyModuleForm.typeId} = ${typeId}`;
+       }
+      const result = await db
+        .select()
+        .from(zuvyModuleForm)
+        .where(
+          and(
+            queryString,
+            sql`((LOWER(${zuvyModuleForm.question}) LIKE '%' || ${searchTerm.toLowerCase()} || '%'))`,
+          ),
+        );
+      return result;
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  async editFormQuestions(editFormDetails: editFormBatchDto) {
+    try {
```

Comment:
To add data validation on the backend side, you can ensure that each option for a question has a length greater than 0. same for the other values.